Index: chrome/app/media_router_strings.grdp
diff --git a/chrome/app/media_router_strings.grdp b/chrome/app/media_router_strings.grdp
index 173135473c3ee5adf2aa8653c7c8a83510b82b05..87417209d36eb5dfa5152eb041330fc1982bd785 100644
--- a/chrome/app/media_router_strings.grdp
+++ b/chrome/app/media_router_strings.grdp
@@ -11,6 +11,9 @@
   <message name="IDS_MEDIA_ROUTER_MENU_ITEM_TITLE" desc="Title of menu item for Media Router, which appears in the overflow menu and page contextual menus.">
     Cast...
   </message>
+  <message name="IDS_MEDIA_ROUTER_LEARN_MORE" desc="Text of a menu item or link which, on click, opens a page with more information or relevant help center page.">
+   Learn more
+  </message>
 
   <!-- Cast Modes -->
   <message name="IDS_MEDIA_ROUTER_AUTO_CAST_MODE" desc="Title for auto cast mode that appears as the header of a list of devices. This represents the mode that will be initially shown, if the list of sinks supports more than one of the cast modes below. Note that auto mode is not a real cast mode and merely serves as a proxy for the other cast modes. This is NOT shown as a dropdown option.">
@@ -41,9 +44,6 @@
   <message name="IDS_MEDIA_ROUTER_HELP" desc="Title of a menu item which, on click, opens a page to the Chromecast help center.">
    Help
   </message>
-  <message name="IDS_MEDIA_ROUTER_LEARN_MORE" desc="Title of a menu item which, on click, opens a page with instructions on how to cast to a Chromecast.">
-   Learn more
-  </message>
   <message name="IDS_MEDIA_ROUTER_REPORT_ISSUE" desc="Title of a menu item which, on click, opens a page to give feedback or report an issue with Chromecast integration.">
    Report an issue
   </message>
@@ -79,9 +79,6 @@
   <message name="IDS_MEDIA_ROUTER_DISMISS_BUTTON" desc="Dismiss button, which, on click, will dismiss the current issue.">
     Dismiss
   </message>
-  <message name="IDS_MEDIA_ROUTER_LEARN_MORE_BUTTON" desc="Learn more button, which, on click, will open a page with more details about the current issue.">
-    Learn More
-  </message>
 
   <!-- Issue Messaging -->
   <message name="IDS_MEDIA_ROUTER_ISSUE_HEADER" desc="Title of the header in the dialog when there is a user action blocking issue surfaced.">

Index: chrome/browser/resources/media_router/elements/issue_banner/issue_banner.js
diff --git a/chrome/browser/resources/media_router/elements/issue_banner/issue_banner.js b/chrome/browser/resources/media_router/elements/issue_banner/issue_banner.js
index 029568514cb4b56afef2dc28eaddff2a5384e830..f897b0e41d13a7a6610796f8864c77b36ba26fbb 100644
--- a/chrome/browser/resources/media_router/elements/issue_banner/issue_banner.js
+++ b/chrome/browser/resources/media_router/elements/issue_banner/issue_banner.js
@@ -19,7 +19,7 @@ Polymer({
       type: Array,
       readOnly: true,
       value: function() {
-        return ['dismissButton', 'learnMoreButton'];
+        return ['dismissButton', 'learnMoreText'];
       },
     },
 
Index: chrome/browser/resources/media_router/elements/media_router_container/media_router_container.js
diff --git a/chrome/browser/resources/media_router/elements/media_router_container/media_router_container.js b/chrome/browser/resources/media_router/elements/media_router_container/media_router_container.js
index 4a2bd33a44766bb793c8cd635ef61e7483b212e5..0fc28c51644355092f0fb4fd1762ad295d122039 100644
--- a/chrome/browser/resources/media_router/elements/media_router_container/media_router_container.js
+++ b/chrome/browser/resources/media_router/elements/media_router_container/media_router_container.js
@@ -115,8 +115,7 @@ Polymer({
     firstRunFlowLearnMore_: {
       type: String,
       readOnly: true,
-      value: loadTimeData.valueExists('firstRunFlowLearnMore') ?
-          loadTimeData.getString('firstRunFlowLearnMore') : '',
+      value: loadTimeData.getString('learnMoreText'),
     },
 
     /**
Index: chrome/browser/ui/webui/media_router/media_router_localized_strings_provider.cc
diff --git a/chrome/browser/ui/webui/media_router/media_router_localized_strings_provider.cc b/chrome/browser/ui/webui/media_router/media_router_localized_strings_provider.cc
index a8778b6a2a08cf67555fecdc415e46b6c6ff3b05..db87c1a0fa7edb03ec87eb1e8201ff3d16ee16df 100644
--- a/chrome/browser/ui/webui/media_router/media_router_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/media_router/media_router_localized_strings_provider.cc
@@ -15,6 +15,8 @@ const char kLocalizedStringsFile[] = "strings.js";
 
 void AddMediaRouterStrings(content::WebUIDataSource* html_source) {
   html_source->AddLocalizedString("mediaRouterTitle", IDS_MEDIA_ROUTER_TITLE);
+  html_source->AddLocalizedString("learnMoreText",
+                                  IDS_MEDIA_ROUTER_LEARN_MORE);
 }
 
 void AddRouteDetailsStrings(content::WebUIDataSource* html_source) {
@@ -31,15 +33,11 @@ void AddIssuesStrings(content::WebUIDataSource* html_source) {
                                   IDS_MEDIA_ROUTER_DISMISS_BUTTON);
   html_source->AddLocalizedString("issueHeader",
                                   IDS_MEDIA_ROUTER_ISSUE_HEADER);
-  html_source->AddLocalizedString("learnMoreButton",
-                                  IDS_MEDIA_ROUTER_LEARN_MORE_BUTTON);
 }
 
 void AddMediaRouterContainerStrings(content::WebUIDataSource* html_source) {
   html_source->AddLocalizedString("firstRunFlowButton",
                                   IDS_MEDIA_ROUTER_FIRST_RUN_FLOW_BUTTON);
-  html_source->AddLocalizedString("firstRunFlowLearnMore",
-      IDS_MEDIA_ROUTER_FIRST_RUN_FLOW_CLOUD_LEARN_MORE);
   html_source->AddLocalizedString("firstRunFlowText",
                                   IDS_MEDIA_ROUTER_FIRST_RUN_FLOW_TEXT);
   html_source->AddLocalizedString("firstRunFlowTitle",

