diff -up chromium-63.0.3239.84/cc/blink/web_layer_impl.h.setopaque chromium-63.0.3239.84/cc/blink/web_layer_impl.h
--- chromium-63.0.3239.84/cc/blink/web_layer_impl.h.setopaque	2017-12-07 10:47:48.262446682 -0500
+++ chromium-63.0.3239.84/cc/blink/web_layer_impl.h	2017-12-07 10:48:47.437298948 -0500
@@ -67,7 +67,7 @@ class CC_BLINK_EXPORT WebLayerImpl : pub
   void SetIsRootForIsolatedGroup(bool root) override;
   bool IsRootForIsolatedGroup() override;
   void SetHitTestableWithoutDrawsContent(bool should_hit_test) override;
-  void SetOpaque(bool opaque) override;
+  CC_BLINK_EXPORT void SetOpaque(bool opaque) override;
   bool Opaque() const override;
   void SetPosition(const blink::WebFloatPoint& position) override;
   blink::WebFloatPoint GetPosition() const override;
