diff -up chromium-66.0.3359.139/skia/ext/convolver_neon.cc.arm-init-fix chromium-66.0.3359.139/skia/ext/convolver_neon.cc
--- chromium-66.0.3359.139/skia/ext/convolver_neon.cc.arm-init-fix	2018-05-05 10:07:03.624424987 -0400
+++ chromium-66.0.3359.139/skia/ext/convolver_neon.cc	2018-05-05 10:07:45.417503001 -0400
@@ -23,7 +23,7 @@ AccumRemainder(const unsigned char* pixe
     remainder[2] += coeff * pixels_left[i * 4 + 2];
     remainder[3] += coeff * pixels_left[i * 4 + 3];
   }
-  return {remainder[0], remainder[1], remainder[2], remainder[3]};
+  return vld1q_s32(remainder);
 }
 
 // Convolves horizontally along a single row. The row data is given in
