diff -up chromium-53.0.2785.101/sandbox/linux/BUILD.gn.415208 chromium-53.0.2785.101/sandbox/linux/BUILD.gn
--- chromium-53.0.2785.101/sandbox/linux/BUILD.gn.415208	2016-09-08 10:22:07.702555383 -0400
+++ chromium-53.0.2785.101/sandbox/linux/BUILD.gn	2016-09-08 10:23:26.190044992 -0400
@@ -41,10 +41,7 @@ group("sandbox") {
     public_deps += [ ":suid_sandbox_client" ]
   }
   if (use_seccomp_bpf || is_nacl_nonsfi) {
-    public_deps += [
-      ":seccomp_bpf",
-      ":seccomp_bpf_helpers",
-    ]
+    public_deps += [ ":seccomp_bpf" ]
   }
 }
 
@@ -221,6 +218,14 @@ component("seccomp_bpf") {
     "bpf_dsl/syscall_set.cc",
     "bpf_dsl/syscall_set.h",
     "bpf_dsl/trap_registry.h",
+    "seccomp-bpf-helpers/baseline_policy.cc",
+    "seccomp-bpf-helpers/baseline_policy.h",
+    "seccomp-bpf-helpers/sigsys_handlers.cc",
+    "seccomp-bpf-helpers/sigsys_handlers.h",
+    "seccomp-bpf-helpers/syscall_parameters_restrictions.cc",
+    "seccomp-bpf-helpers/syscall_parameters_restrictions.h",
+    "seccomp-bpf-helpers/syscall_sets.cc",
+    "seccomp-bpf-helpers/syscall_sets.h",
     "seccomp-bpf/die.cc",
     "seccomp-bpf/die.h",
     "seccomp-bpf/sandbox_bpf.cc",
@@ -250,31 +255,6 @@ component("seccomp_bpf") {
       "bpf_dsl/linux_syscall_ranges.h",
       "bpf_dsl/seccomp_macros.h",
       "bpf_dsl/trap_registry.h",
-    ]
-  }
-}
-
-component("seccomp_bpf_helpers") {
-  sources = [
-    "seccomp-bpf-helpers/baseline_policy.cc",
-    "seccomp-bpf-helpers/baseline_policy.h",
-    "seccomp-bpf-helpers/sigsys_handlers.cc",
-    "seccomp-bpf-helpers/sigsys_handlers.h",
-    "seccomp-bpf-helpers/syscall_parameters_restrictions.cc",
-    "seccomp-bpf-helpers/syscall_parameters_restrictions.h",
-    "seccomp-bpf-helpers/syscall_sets.cc",
-    "seccomp-bpf-helpers/syscall_sets.h",
-  ]
-  defines = [ "SANDBOX_IMPLEMENTATION" ]
-
-  deps = [
-    ":sandbox_services",
-    ":seccomp_bpf",
-    "//base",
-  ]
-
-  if (is_nacl_nonsfi) {
-    sources -= [
       "seccomp-bpf-helpers/baseline_policy.cc",
       "seccomp-bpf-helpers/baseline_policy.h",
       "seccomp-bpf-helpers/syscall_sets.cc",
diff -up chromium-53.0.2785.101/sandbox/linux/sandbox_linux.gypi.415208 chromium-53.0.2785.101/sandbox/linux/sandbox_linux.gypi
--- chromium-53.0.2785.101/sandbox/linux/sandbox_linux.gypi.415208	2016-09-08 10:20:50.263045995 -0400
+++ chromium-53.0.2785.101/sandbox/linux/sandbox_linux.gypi	2016-09-08 10:21:51.341870321 -0400
@@ -54,7 +54,6 @@
         [ 'use_seccomp_bpf==1', {
           'dependencies': [
             'seccomp_bpf',
-            'seccomp_bpf_helpers',
           ],
         }],
       ],
@@ -141,6 +140,14 @@
         'bpf_dsl/syscall_set.cc',
         'bpf_dsl/syscall_set.h',
         'bpf_dsl/trap_registry.h',
+        'seccomp-bpf-helpers/baseline_policy.cc',
+        'seccomp-bpf-helpers/baseline_policy.h',
+        'seccomp-bpf-helpers/sigsys_handlers.cc',
+        'seccomp-bpf-helpers/sigsys_handlers.h',
+        'seccomp-bpf-helpers/syscall_parameters_restrictions.cc',
+        'seccomp-bpf-helpers/syscall_parameters_restrictions.h',
+        'seccomp-bpf-helpers/syscall_sets.cc',
+        'seccomp-bpf-helpers/syscall_sets.h',
         'seccomp-bpf/die.cc',
         'seccomp-bpf/die.h',
         'seccomp-bpf/sandbox_bpf.cc',
@@ -165,31 +172,6 @@
       ],
       'include_dirs': [
         '../..',
-      ],
-    },
-    {
-      'target_name': 'seccomp_bpf_helpers',
-      'type': '<(component)',
-      'sources': [
-        'seccomp-bpf-helpers/baseline_policy.cc',
-        'seccomp-bpf-helpers/baseline_policy.h',
-        'seccomp-bpf-helpers/sigsys_handlers.cc',
-        'seccomp-bpf-helpers/sigsys_handlers.h',
-        'seccomp-bpf-helpers/syscall_parameters_restrictions.cc',
-        'seccomp-bpf-helpers/syscall_parameters_restrictions.h',
-        'seccomp-bpf-helpers/syscall_sets.cc',
-        'seccomp-bpf-helpers/syscall_sets.h',
-      ],
-      'dependencies': [
-        '../base/base.gyp:base',
-        'sandbox_services',
-        'seccomp_bpf',
-      ],
-      'defines': [
-        'SANDBOX_IMPLEMENTATION',
-      ],
-      'include_dirs': [
-        '../..',
       ],
     },
     {
diff -up chromium-53.0.2785.101/services/shell/runner/host/BUILD.gn.415208 chromium-53.0.2785.101/services/shell/runner/host/BUILD.gn
--- chromium-53.0.2785.101/services/shell/runner/host/BUILD.gn.415208	2016-09-08 10:20:17.591674773 -0400
+++ chromium-53.0.2785.101/services/shell/runner/host/BUILD.gn	2016-09-08 10:20:27.742478927 -0400
@@ -61,7 +61,6 @@ source_set("child_process_base") {
       "//sandbox/linux:sandbox",
       "//sandbox/linux:sandbox_services",
       "//sandbox/linux:seccomp_bpf",
-      "//sandbox/linux:seccomp_bpf_helpers",
     ]
   }
 
diff -up chromium-53.0.2785.101/services/shell/shell.gyp.415208 chromium-53.0.2785.101/services/shell/shell.gyp
--- chromium-53.0.2785.101/services/shell/shell.gyp.415208	2016-09-08 10:19:52.543156283 -0400
+++ chromium-53.0.2785.101/services/shell/shell.gyp	2016-09-08 10:20:03.894938333 -0400
@@ -145,7 +145,6 @@
           '<(DEPTH)/sandbox/sandbox.gyp:sandbox',
           '<(DEPTH)/sandbox/sandbox.gyp:sandbox_services',
           '<(DEPTH)/sandbox/sandbox.gyp:seccomp_bpf',
-          '<(DEPTH)/sandbox/sandbox.gyp:seccomp_bpf_helpers',
         ],
       }],
       ['OS=="mac"', {
