Index: chromium-71.0.3578.80/media/gpu/vaapi/vaapi_jpeg_decode_accelerator.cc
===================================================================
--- chromium-71.0.3578.80.orig/media/gpu/vaapi/vaapi_jpeg_decode_accelerator.cc
+++ chromium-71.0.3578.80/media/gpu/vaapi/vaapi_jpeg_decode_accelerator.cc
@@ -47,6 +47,9 @@ static void ReportToUMA(VAJDADecoderFail
 // macro in the header file without pulling in the entire <va/va.h>.
 static_assert(VA_FOURCC_YUYV == VA_FOURCC('Y', 'U', 'Y', 'V'),
               "VA_FOURCC_YUYV must be equal to VA_FOURCC('Y', 'U', 'Y', 'V')");
+#ifndef VA_FOURCC_I420
+#define VA_FOURCC_I420 0x30323449
+#endif
 constexpr VAImageFormat kImageFormatI420 = {.fourcc = VA_FOURCC_I420,
                                             .byte_order = VA_LSB_FIRST,
                                             .bits_per_pixel = 12};
