Index: remoting/host/fake_desktop_environment.cc
diff --git a/remoting/host/fake_desktop_environment.cc b/remoting/host/fake_desktop_environment.cc
index d566b7a5d241220c2d7f824c4b3b8cfa157a7f43..bede5ee2d476b29ee8aeb3e77fa36add82de23f1 100644
--- a/remoting/host/fake_desktop_environment.cc
+++ b/remoting/host/fake_desktop_environment.cc
@@ -85,7 +85,7 @@ FakeDesktopEnvironment::CreateVideoCapturer() {
   std::unique_ptr<DesktopCapturerProxy> result(
       new DesktopCapturerProxy(capture_thread_));
   result->set_capturer(std::move(fake_capturer));
-  return result;
+  return std::move(result);
 }
 
 std::unique_ptr<webrtc::MouseCursorMonitor>
