diff -urN chromium-20.0.1132.47.orig/third_party/tcmalloc/chromium/src/base/linuxthreads.cc chromium-20.0.1132.47/third_party/tcmalloc/chromium/src/base/linuxthreads.cc
--- chromium-20.0.1132.47.orig/third_party/tcmalloc/chromium/src/base/linuxthreads.cc	2012-07-10 12:32:14.572427919 +0400
+++ chromium-20.0.1132.47/third_party/tcmalloc/chromium/src/base/linuxthreads.cc	2012-07-10 12:34:09.725538925 +0400
@@ -193,7 +193,7 @@
 /* Signal handler to help us recover from dying while we are attached to
  * other threads.
  */
-static void SignalHandler(int signum, siginfo_t *si, void *data) {
+static void SignalHandler(int signum, siginfo *si, void *data) {
   if (sig_pids != NULL) {
     if (signum == SIGABRT) {
       while (sig_num_threads-- > 0) {
