--- chromium-20100218svn39394/src/chrome/browser/sync/util/sqlite_utils.cc.no-sqlite-debugging	2010-02-18 18:23:31.000000000 -0500
+++ chromium-20100218svn39394/src/chrome/browser/sync/util/sqlite_utils.cc	2010-02-19 15:40:50.338131672 -0500
@@ -33,9 +33,6 @@ class DebugSQLErrorHandler: public Vanil
  public:
   virtual int HandleError(int error, sqlite3* db) {
     error_ = error;
-    LOG(ERROR) << "sqlite error " << error
-               << " " << sqlite3_errmsg(db)
-               << " db " << static_cast<void*>(db);
     return error;
   }
 };
@@ -281,8 +279,7 @@ int SQLStatement::prepare(sqlite3* db, c
   DCHECK(!stmt_);
   int rv = sqlite3_prepare_v2(db, sql, sql_len, &stmt_, NULL);
   if (rv != SQLITE_OK) {
-    SQLErrorHandler* error_handler = GetErrorHandlerFactory()->Make();
-    return error_handler->HandleError(rv, db);
+    DLOG(ERROR) << "SQLStatement.prepare_v2 failed: " << sqlite3_errmsg(db);
   }
   return rv;
 }
