diff -up chromium-62.0.3202.62/tools/gn/bootstrap/bootstrap.py.epel7-noc++14 chromium-62.0.3202.62/tools/gn/bootstrap/bootstrap.py
--- chromium-62.0.3202.62/tools/gn/bootstrap/bootstrap.py.epel7-noc++14	2017-10-24 22:20:25.881183458 -0400
+++ chromium-62.0.3202.62/tools/gn/bootstrap/bootstrap.py	2017-10-24 22:20:38.395943374 -0400
@@ -349,7 +349,7 @@ def write_gn_ninja(path, root_gen_dir, o
         '-pipe',
         '-fno-exceptions'
     ])
-    cflags_cc.extend(['-std=c++14', '-Wno-c++11-narrowing'])
+    cflags_cc.extend(['-std=c++11', '-Wno-c++11-narrowing'])
     if is_aix:
      cflags.extend(['-maix64'])
      ldflags.extend([ '-maix64 -Wl,-bbigtoc' ])
