diff -up chromium-52.0.2743.82/sandbox/linux/sandbox_linux.gypi.pie chromium-52.0.2743.82/sandbox/linux/sandbox_linux.gypi
--- chromium-52.0.2743.82/sandbox/linux/sandbox_linux.gypi.pie	2016-08-03 09:06:41.617923620 +0200
+++ chromium-52.0.2743.82/sandbox/linux/sandbox_linux.gypi	2016-08-03 09:11:49.747463317 +0200
@@ -205,7 +205,10 @@
       ],
       'cflags': [
         # For ULLONG_MAX
-        '-std=gnu99',
+        '-std=gnu99 -fPIE',
+      ],
+      'ldflags': [
+        '-pie',
       ],
       'include_dirs': [
         '../..',
