Index: chromium/src/build/all.gyp
===================================================================
--- chromium.orig/src/build/all.gyp
+++ chromium/src/build/all.gyp
@@ -51,7 +51,7 @@
         '../third_party/WebKit/Source/WebKit/chromium/All.gyp:*',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:generate_devtools_zip',
         '../third_party/zlib/zlib.gyp:*',
-        '../v8/tools/gyp/v8.gyp:*',
+        'linux/system.gyp:v8',
         '../webkit/support/webkit_support.gyp:*',
         '../webkit/webkit.gyp:*',
         'util/build_util.gyp:*',
Index: chromium/src/build/linux/system.gyp
===================================================================
--- chromium.orig/src/build/linux/system.gyp
+++ chromium/src/build/linux/system.gyp
@@ -380,6 +380,19 @@
       ],
     },
     {
+      'target_name': 'v8',
+      'type': 'none',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'libraries': [
+              '-lv8',
+              '-lvpx',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome_keyring',
       'type': 'none',
       'conditions': [
Index: chromium/src/net/net.gyp
===================================================================
--- chromium.orig/src/net/net.gyp
+++ chromium/src/net/net.gyp
@@ -35,7 +35,7 @@
         '../third_party/icu/icu.gyp:icui18n',
         '../third_party/icu/icu.gyp:icuuc',
         '../third_party/zlib/zlib.gyp:zlib',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         'net_resources',
         'ssl_false_start_blacklist_process#host',
       ],
Index: chromium/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
===================================================================
--- chromium.orig/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
+++ chromium/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
@@ -1070,7 +1070,7 @@
         '<(chromium_src_dir)/third_party/libwebp/libwebp.gyp:libwebp',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       'include_dirs': [
@@ -1189,7 +1189,7 @@
         '<(chromium_src_dir)/third_party/ots/ots.gyp:ots',
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       'export_dependent_settings': [
@@ -1207,7 +1207,7 @@
         '<(chromium_src_dir)/third_party/ots/ots.gyp:ots',
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       # This is needed for mac because of webkit_system_interface. It'd be nice
@@ -1967,7 +1967,7 @@
         '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
         '<(chromium_src_dir)/skia/skia.gyp:skia',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
       ],
       'export_dependent_settings': [
         'webcore_bindings',
@@ -1976,7 +1976,7 @@
         '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
         '<(chromium_src_dir)/skia/skia.gyp:skia',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
