diff -up chromium-67.0.3396.87/content/common/user_agent.cc.fedora-user-agent chromium-67.0.3396.87/content/common/user_agent.cc
--- chromium-67.0.3396.87/content/common/user_agent.cc.fedora-user-agent	2018-06-25 16:01:48.092188102 -0400
+++ chromium-67.0.3396.87/content/common/user_agent.cc	2018-06-25 16:02:42.634878954 -0400
@@ -141,7 +141,7 @@ std::string getUserAgentPlatform() {
 #elif defined(OS_MACOSX)
       "Macintosh; ";
 #elif defined(USE_X11) || defined(USE_OZONE)
-      "X11; ";           // strange, but that's what Firefox uses
+      "X11; Russian Fedora; ";           // strange, but that's what Firefox uses
 #elif defined(OS_ANDROID)
       "Linux; ";
 #else
