diff -up chromium-54.0.2840.90/cc/blink/web_layer_impl.h.setopaque chromium-54.0.2840.90/cc/blink/web_layer_impl.h
--- chromium-54.0.2840.90/cc/blink/web_layer_impl.h.setopaque	2016-11-03 09:07:03.422497483 -0400
+++ chromium-54.0.2840.90/cc/blink/web_layer_impl.h	2016-11-03 09:07:13.302307531 -0400
@@ -80,7 +80,7 @@ class WebLayerImpl : public blink::WebLa
   blink::WebBlendMode blendMode() const override;
   void setIsRootForIsolatedGroup(bool root) override;
   bool isRootForIsolatedGroup() override;
-  void setOpaque(bool opaque) override;
+  CC_BLINK_EXPORT void setOpaque(bool opaque) override;
   bool opaque() const override;
   void setPosition(const blink::WebFloatPoint& position) override;
   blink::WebFloatPoint position() const override;
diff -up chromium-54.0.2840.90/content/renderer/child_frame_compositing_helper.cc.setopaque chromium-54.0.2840.90/content/renderer/child_frame_compositing_helper.cc
