Index: chromium/src/third_party/libjingle/source/talk/xmllite/xmlbuilder.h
===================================================================
--- chromium.orig/src/third_party/libjingle/source/talk/xmllite/xmlbuilder.h
+++ chromium/src/third_party/libjingle/source/talk/xmllite/xmlbuilder.h
@@ -33,11 +33,7 @@
 #include "talk/base/scoped_ptr.h"
 #include "talk/xmllite/xmlparser.h"
 
-#ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
-#else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
-#endif  // EXPAT_RELATIVE_PATH
+#include <expat.h>
 
 namespace buzz {
 
Index: chromium/src/third_party/libjingle/source/talk/xmllite/xmlparser.h
===================================================================
--- chromium.orig/src/third_party/libjingle/source/talk/xmllite/xmlparser.h
+++ chromium/src/third_party/libjingle/source/talk/xmllite/xmlparser.h
@@ -31,11 +31,7 @@
 #include <string>
 
 #include "talk/xmllite/xmlnsstack.h"
-#ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
-#else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
-#endif  // EXPAT_RELATIVE_PATH
+#include <expat.h>
 
 struct XML_ParserStruct;
 typedef struct XML_ParserStruct* XML_Parser;
