diff -up chromium/src/net/base/gzip_header.cc chromium/src/net/base/gzip_header.cc
--- chromium/src/net/base/gzip_header.cc        2009-09-28 10:34:28.000000000 -0600
+++ chromium/src/net/base/gzip_header.cc    2009-11-24 17:23:22.000000000 -0700
@@ -4,11 +4,11 @@

 #include "net/base/gzip_header.h"

-#if defined(USE_SYSTEM_ZLIB)
+/* #if defined(USE_SYSTEM_ZLIB) */
 #include <zlib.h>
-#else
+/* #else
 #include "third_party/zlib/zlib.h"  // for Z_DEFAULT_COMPRESSION
-#endif
+#endif */

 #include "base/logging.h"

diff -up chromium/src/net/base/sdch_filter_unittest.cc chromium/src/net/base/sdch_filter_unittest.cc
--- chromium/src/net/base/sdch_filter_unittest.cc       2009-09-28 10:34:29.000000000 -0600
+++ chromium/src/net/base/sdch_filter_unittest.cc   2009-11-24 17:21:57.000000000 -0700
@@ -8,11 +8,11 @@
 #include <string>
 #include <vector>

-#if defined(USE_SYSTEM_ZLIB)
+/* #if defined(USE_SYSTEM_ZLIB) */
 #include <zlib.h>
-#else
+/* #else
 #include "third_party/zlib/zlib.h"
-#endif
+#endif */

 #include "base/logging.h"
 #include "base/memory/scoped_ptr.h"
