diff -up chromium-59.0.3071.86/build/config/clang/clang.gni.system-clang chromium-59.0.3071.86/build/config/clang/clang.gni
--- chromium-59.0.3071.86/build/config/clang/clang.gni.system-clang	2017-06-11 20:57:07.327949687 -0400
+++ chromium-59.0.3071.86/build/config/clang/clang.gni	2017-06-11 20:57:24.085620826 -0400
@@ -9,5 +9,5 @@ declare_args() {
   # coding guidelines, etc. Only used when compiling with Clang.
   clang_use_chrome_plugins = is_clang && !is_nacl && !use_xcode_clang
 
-  clang_base_path = "//third_party/llvm-build/Release+Asserts"
+  clang_base_path = "/usr"
 }
