diff -up chromium-65.0.3325.146/components/policy/core/browser/browser_policy_connector_base.h.fully-declare chromium-65.0.3325.146/components/policy/core/browser/browser_policy_connector_base.h
--- chromium-65.0.3325.146/components/policy/core/browser/browser_policy_connector_base.h.fully-declare	2018-03-13 23:02:13.989635567 -0400
+++ chromium-65.0.3325.146/components/policy/core/browser/browser_policy_connector_base.h	2018-03-13 23:02:44.318048625 -0400
@@ -12,13 +12,13 @@
 #include "base/macros.h"
 #include "base/optional.h"
 #include "components/policy/core/browser/configuration_policy_handler_list.h"
+#include "components/policy/core/common/configuration_policy_provider.h"
 #include "components/policy/core/common/schema.h"
 #include "components/policy/core/common/schema_registry.h"
 #include "components/policy/policy_export.h"
 
 namespace policy {
 
-class ConfigurationPolicyProvider;
 class PolicyService;
 class PolicyServiceImpl;
 
