diff -up chromium-52.0.2723.2/sync/sync_tests.gypi.sync_link_zlib chromium-52.0.2723.2/sync/sync_tests.gypi
--- chromium-52.0.2723.2/sync/sync_tests.gypi.sync_link_zlib	2016-05-09 14:22:30.884719562 -0400
+++ chromium-52.0.2723.2/sync/sync_tests.gypi	2016-05-09 14:24:06.337517725 -0400
@@ -264,6 +264,7 @@
         '../testing/gtest.gyp:gtest',
         '../third_party/leveldatabase/leveldatabase.gyp:leveldatabase',
         '../third_party/protobuf/protobuf.gyp:protobuf_lite',
+        '../third_party/zlib/zlib.gyp:zlib',
         'sync',
         'test_support_sync_api',
         'test_support_sync_core',
