diff -urN chromium.orig/src/content/browser/gamepad/platform_data_fetcher.h chromium/src/content/browser/gamepad/platform_data_fetcher.h
--- chromium.orig/src/content/browser/gamepad/platform_data_fetcher.h	2012-03-03 22:37:25.610878628 +0400
+++ chromium/src/content/browser/gamepad/platform_data_fetcher.h	2012-03-03 22:39:31.851571064 +0400
@@ -16,8 +16,6 @@
 #include "content/browser/gamepad/platform_data_fetcher_win.h"
 #elif defined(OS_MACOSX)
 #include "content/browser/gamepad/platform_data_fetcher_mac.h"
-#elif defined(OS_LINUX)
-#include "content/browser/gamepad/platform_data_fetcher_linux.h"
 #endif
 
 namespace content {
@@ -30,10 +28,6 @@
 
 typedef GamepadPlatformDataFetcherMac GamepadPlatformDataFetcher;
 
-#elif defined(OS_LINUX)
-
-typedef GamepadPlatformDataFetcherLinux GamepadPlatformDataFetcher;
-
 #else
 
 class GamepadDataFetcherEmpty : public GamepadDataFetcher {
diff -urN chromium.orig/src/content/content_browser.gypi chromium/src/content/content_browser.gypi
--- chromium.orig/src/content/content_browser.gypi	2012-03-03 22:37:25.610878628 +0400
+++ chromium/src/content/content_browser.gypi	2012-03-03 22:39:02.525525637 +0400
@@ -690,7 +690,7 @@
     '<(SHARED_INTERMEDIATE_DIR)/webkit/grit/devtools_resources_map.h',
   ],
   'conditions': [
-    ['OS!="win" and OS!="mac" and OS!="linux"', {
+    ['OS=="linux"', {
       'sources': [
         'browser/gamepad/platform_data_fetcher.cc',
       ]
@@ -743,9 +743,6 @@
       ],
     }],
     ['OS=="linux"', {
-      'dependencies': [
-        '../build/linux/system.gyp:udev',
-      ],
       # Set LINUX_SANDBOX_PATH for Linux, including Chrome OS
       'conditions': [
         ['linux_sandbox_path != ""', {
@@ -797,6 +794,11 @@
         'browser/renderer_host/gtk_key_bindings_handler.h',
       ],
     }],
+    ['OS=="linux"', {
+      'sources/': [
+        ['exclude', '^browser/gamepad/platform_data_fetcher_linux\\.cc$'],
+      ],
+    }],
     ['os_bsd==1', {
       'sources/': [
         ['exclude', '^browser/gamepad/platform_data_fetcher_linux\\.cc$'],
