diff -urN chromium-44.0.2403.89.orig/third_party/widevine/cdm/widevine_cdm.gyp chromium-44.0.2403.89/third_party/widevine/cdm/widevine_cdm.gyp
--- chromium-44.0.2403.89.orig/third_party/widevine/cdm/widevine_cdm.gyp	2015-07-23 12:28:47.187041283 +0300
+++ chromium-44.0.2403.89/third_party/widevine/cdm/widevine_cdm.gyp	2015-07-23 12:47:17.785797088 +0300
@@ -12,7 +12,7 @@
     'widevine_cdm_version_h_file%': 'widevine_cdm_version.h',
     'widevine_cdm_binary_files%': [],
     'conditions': [
-      [ 'branding == "Chrome"', {
+      [ 'branding == "Chromium"', {
         'conditions': [
           [ 'chromeos == 1', {
             'widevine_cdm_version_h_file%':
@@ -88,7 +88,7 @@
       'target_name': 'widevinecdmadapter',
       'type': 'none',
       'conditions': [
-        [ '(branding == "Chrome" or enable_widevine == 1) and enable_pepper_cdms == 1', {
+        [ '(branding == "Chromium" or enable_widevine == 1) and enable_pepper_cdms == 1', {
           'dependencies': [
             '<(DEPTH)/ppapi/ppapi.gyp:ppapi_cpp',
             '<(DEPTH)/media/media_cdm_adapter.gyp:cdmadapter',
diff -urN chromium-44.0.2403.89.orig/third_party/widevine/cdm/widevine_cdm_version.h chromium-44.0.2403.89/third_party/widevine/cdm/widevine_cdm_version.h
--- chromium-44.0.2403.89.orig/third_party/widevine/cdm/widevine_cdm_version.h	2015-07-23 12:28:47.187041283 +0300
+++ chromium-44.0.2403.89/third_party/widevine/cdm/widevine_cdm_version.h	2015-07-23 12:34:34.716387100 +0300
@@ -14,4 +14,8 @@
 //  - WIDEVINE_CDM_VERSION_STRING (with the version of the CDM that's available
 //    as a string, e.g., "1.0.123.456").
 
+#include "third_party/widevine/cdm/widevine_cdm_common.h"
+#define WIDEVINE_CDM_AVAILABLE
+#define WIDEVINE_CDM_VERSION_STRING "@WIDEVINE_VERSION@"
+
 #endif  // WIDEVINE_CDM_VERSION_H_
