diff -urN chromium-61.0.3128.3.orig/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h chromium-61.0.3128.3/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h
--- chromium-61.0.3128.3.orig/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h	2017-06-19 12:30:37.581788368 +0300
+++ chromium-61.0.3128.3/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h	2017-06-19 12:31:45.866657857 +0300
@@ -8,6 +8,7 @@
 #include "platform/PlatformExport.h"
 #include "platform/wtf/ThreadSpecific.h"
 
+#include <functional>
 #include <memory>
 
 namespace gpu {
diff -urN chromium-61.0.3128.3.orig/v8/src/objects-body-descriptors.h chromium-61.0.3128.3/v8/src/objects-body-descriptors.h
--- chromium-61.0.3128.3.orig/v8/src/objects-body-descriptors.h	2017-06-19 12:30:22.561596255 +0300
+++ chromium-61.0.3128.3/v8/src/objects-body-descriptors.h	2017-06-19 12:33:01.379619613 +0300
@@ -99,7 +99,7 @@
 
   template <typename StaticVisitor>
   static inline void IterateBody(HeapObject* obj, int object_size) {
-    IterateBody(obj);
+    IterateBody<StaticVisitor>(obj);
   }
 
   static inline int SizeOf(Map* map, HeapObject* object) { return kSize; }
