diff -up chromium-65.0.3325.146/services/preferences/tracked/pref_hash_filter.h.fulldecl chromium-65.0.3325.146/services/preferences/tracked/pref_hash_filter.h
--- chromium-65.0.3325.146/services/preferences/tracked/pref_hash_filter.h.fulldecl	2018-03-13 16:38:38.870652491 -0400
+++ chromium-65.0.3325.146/services/preferences/tracked/pref_hash_filter.h	2018-03-13 16:39:02.691186647 -0400
@@ -21,9 +21,9 @@
 #include "services/preferences/public/interfaces/preferences.mojom.h"
 #include "services/preferences/tracked/hash_store_contents.h"
 #include "services/preferences/tracked/interceptable_pref_filter.h"
+#include "services/preferences/tracked/pref_hash_store.h"
 #include "services/preferences/tracked/tracked_preference.h"
 
-class PrefHashStore;
 class PrefService;
 
 namespace base {
diff -up chromium-65.0.3325.146/third_party/WebKit/Source/modules/webdatabase/SQLTransactionBackend.h.fulldecl chromium-65.0.3325.146/third_party/WebKit/Source/modules/webdatabase/SQLTransactionBackend.h
--- chromium-65.0.3325.146/third_party/WebKit/Source/modules/webdatabase/SQLTransactionBackend.h.fulldecl	2018-03-13 16:39:13.787970233 -0400
+++ chromium-65.0.3325.146/third_party/WebKit/Source/modules/webdatabase/SQLTransactionBackend.h	2018-03-13 16:39:42.614407235 -0400
@@ -30,6 +30,7 @@
 
 #include <memory>
 #include "modules/webdatabase/DatabaseBasicTypes.h"
+#include "modules/webdatabase/SQLError.h"
 #include "modules/webdatabase/SQLStatement.h"
 #include "modules/webdatabase/SQLStatementBackend.h"
 #include "modules/webdatabase/SQLTransactionStateMachine.h"
@@ -41,7 +42,6 @@
 namespace blink {
 
 class Database;
-class SQLErrorData;
 class SQLiteTransaction;
 class SQLTransaction;
 class SQLTransactionBackend;
