diff -urN chromium-48.0.2564.103.orig/third_party/skia/src/ports/SkFontHost_fontconfig.cpp chromium-48.0.2564.103/third_party/skia/src/ports/SkFontHost_fontconfig.cpp
--- chromium-48.0.2564.103.orig/third_party/skia/src/ports/SkFontHost_fontconfig.cpp	2016-02-08 17:12:04.382043761 +0300
+++ chromium-48.0.2564.103/third_party/skia/src/ports/SkFontHost_fontconfig.cpp	2016-02-08 17:13:32.625349457 +0300
@@ -11,6 +11,7 @@
 #include "SkStream.h"
 #include "SkTypeface.h"
 #include "SkTypefaceCache.h"
+#include <unistd.h>
 
 ///////////////////////////////////////////////////////////////////////////////
 ///////////////////////////////////////////////////////////////////////////////
