diff -urN chromium-56.0.2924.59.orig/components/strings/components_strings_ru.xtb chromium-56.0.2924.59/components/strings/components_strings_ru.xtb
--- chromium-56.0.2924.59.orig/components/strings/components_strings_ru.xtb	2017-01-16 12:13:16.681398255 +0300
+++ chromium-56.0.2924.59/components/strings/components_strings_ru.xtb	2017-01-16 12:33:51.593536335 +0300
@@ -224,7 +224,7 @@
 <translation id="3305707030755673451">Данные были зашифрованы с помощью кодовой фразы <ph name="TIME" />. Введите ее, чтобы начать синхронизацию.</translation>
 <translation id="3329013043687509092">Насыщенность</translation>
 <translation id="333371639341676808">Предотвратить создание дополнительных диалоговых окон на этой странице.</translation>
-<translation id="3338095232262050444">Надежный</translation>
+<translation id="3338095232262050444">Надёжный</translation>
 <translation id="3340978935015468852">настройках</translation>
 <translation id="3345135638360864351">Не удалось отправить пользователю <ph name="NAME" /> запрос на доступ к этому сайту. Повторите попытку.</translation>
 <translation id="3355823806454867987">Изменить настройки прокси-сервера...</translation>
@@ -741,4 +741,4 @@
 <translation id="962701380617707048">Введите срок действия и CVC-код карты <ph name="CREDIT_CARD" /></translation>
 <translation id="969892804517981540">Официальная сборка</translation>
 <translation id="988159990683914416">Сборка для разработчиков</translation>
-</translationbundle>
\ В конце файла нет новой строки
+</translationbundle>
