Index: chromium/src/build/all.gyp
===================================================================
--- chromium.orig/src/build/all.gyp
+++ chromium/src/build/all.gyp
@@ -53,7 +53,7 @@
         '../third_party/WebKit/Source/WebKit/chromium/All.gyp:*',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:generate_devtools_zip',
         '../third_party/zlib/zlib.gyp:*',
-        '../v8/tools/gyp/v8.gyp:*',
+        'linux/system.gyp:v8',
         '../webkit/support/webkit_support.gyp:*',
         '../webkit/webkit.gyp:*',
         'temp_gyp/googleurl.gyp:*',
Index: chromium/src/build/linux/system.gyp
===================================================================
--- chromium.orig/src/build/linux/system.gyp
+++ chromium/src/build/linux/system.gyp
@@ -384,6 +384,19 @@
       ],
     },
     {
+      'target_name': 'v8',
+      'type': 'none',
+      'conditions': [
+        ['_toolset=="target"', {
+          'link_settings': {
+            'libraries': [
+              '-lv8',
+              '-lvpx',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome_keyring',
       'type': 'none',
       'conditions': [
Index: chromium/src/net/net.gyp
===================================================================
--- chromium.orig/src/net/net.gyp
+++ chromium/src/net/net.gyp
@@ -35,7 +35,7 @@
         '../third_party/icu/icu.gyp:icui18n',
         '../third_party/icu/icu.gyp:icuuc',
         '../third_party/zlib/zlib.gyp:zlib',
-        '../v8/tools/gyp/v8.gyp:v8',
+        '../build/linux/system.gyp:v8',
         'net_resources',
       ],
       'sources': [
Index: chromium/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
===================================================================
--- chromium.orig/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
+++ chromium/src/third_party/WebKit/Source/WebCore/WebCore.gyp/WebCore.gyp
@@ -1114,7 +1114,7 @@
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
         '<(chromium_src_dir)/third_party/qcms/qcms.gyp:qcms',
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       'include_dirs': [
@@ -1245,7 +1245,7 @@
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
         '<(chromium_src_dir)/third_party/zlib/zlib.gyp:zlib',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       'export_dependent_settings': [
@@ -1265,7 +1265,7 @@
         '<(chromium_src_dir)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(chromium_src_dir)/third_party/angle/src/build_angle.gyp:translator_glsl',
         '<(chromium_src_dir)/third_party/zlib/zlib.gyp:zlib',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
         '<(libjpeg_gyp_path):libjpeg',
       ],
       # This is needed for mac because of webkit_system_interface. It'd be nice
@@ -2015,7 +2015,7 @@
         '<(chromium_src_dir)/skia/skia.gyp:skia',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
         '<(chromium_src_dir)/third_party/qcms/qcms.gyp:qcms',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
       ],
       'export_dependent_settings': [
         'webcore_bindings',
@@ -2024,7 +2024,7 @@
         '<(chromium_src_dir)/skia/skia.gyp:skia',
         '<(chromium_src_dir)/third_party/npapi/npapi.gyp:npapi',
         '<(chromium_src_dir)/third_party/qcms/qcms.gyp:qcms',
-        '<(chromium_src_dir)/v8/tools/gyp/v8.gyp:v8',
+        '<(chromium_src_dir)/build/linux/system.gyp:v8',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
