diff -up chromium-68.0.3440.106/extensions/browser/api/networking_private/BUILD.gn.fixb chromium-68.0.3440.106/extensions/browser/api/networking_private/BUILD.gn
--- chromium-68.0.3440.106/extensions/browser/api/networking_private/BUILD.gn.fixb	2018-08-09 03:56:36.091622243 -0400
+++ chromium-68.0.3440.106/extensions/browser/api/networking_private/BUILD.gn	2018-08-09 03:57:03.047001659 -0400
@@ -61,7 +61,7 @@ source_set("networking_private") {
               ]
     deps += [ "//chromeos" ]
   } else {
-    not_needed(default_sources)
+    not_needed([ "default_sources" ])
     sources = [
       "networking_private_stubs.cc",
     ]
