--- chromium/src/remoting/remoting.gyp	2010-10-03 15:13:07.746440175 +0200
+++ chromium/src/remoting/remoting.gyp	2010-10-03 15:13:07.746440175 +0200
@@ -284,7 +284,7 @@
         '../ui/ui.gyp:ui',
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
-        '../third_party/libvpx/libvpx.gyp:libvpx',
+        '../build/linux/system.gyp:libvpx',
         '../third_party/protobuf/protobuf.gyp:protobuf_lite',
         '../third_party/zlib/zlib.gyp:zlib',
         '../media/media.gyp:yuv_convert',
