diff -up chromium-60.0.3112.78/tools/gn/BUILD.gn.lastcommit chromium-60.0.3112.78/tools/gn/BUILD.gn
--- chromium-60.0.3112.78/tools/gn/BUILD.gn.lastcommit	2017-07-31 10:42:38.568041044 -0400
+++ chromium-60.0.3112.78/tools/gn/BUILD.gn	2017-07-31 10:47:50.760974587 -0400
@@ -268,7 +268,6 @@ executable("gn") {
 
   deps = [
     ":gn_lib",
-    ":last_commit_position",
     "//base",
     "//build/config:exe_and_shlib_deps",
     "//build/win:default_exe_manifest",
diff -up chromium-60.0.3112.78/tools/gn/gn_main.cc.lastcommit chromium-60.0.3112.78/tools/gn/gn_main.cc
--- chromium-60.0.3112.78/tools/gn/gn_main.cc.lastcommit	2017-07-25 15:05:19.000000000 -0400
+++ chromium-60.0.3112.78/tools/gn/gn_main.cc	2017-07-31 10:42:38.568041044 -0400
@@ -12,13 +12,7 @@
 #include "tools/gn/standard_out.h"
 #include "tools/gn/switches.h"
 
-// Only the GN-generated build makes this header for now.
-// TODO(brettw) consider adding this if we need it in GYP.
-#if defined(GN_BUILD)
-#include "tools/gn/last_commit_position.h"
-#else
 #define LAST_COMMIT_POSITION "UNKNOWN"
-#endif
 
 namespace {
 
