diff -urN chromium-28.0.1500.45.orig/net/third_party/nss/ssl/bodge/secitem_array.c chromium-28.0.1500.45.new/net/third_party/nss/ssl/bodge/secitem_array.c
--- chromium-28.0.1500.45.orig/net/third_party/nss/ssl/bodge/secitem_array.c	2013-06-20 19:27:52.000000000 +0400
+++ chromium-28.0.1500.45.new/net/third_party/nss/ssl/bodge/secitem_array.c	2013-06-20 19:49:37.120194892 +0400
@@ -11,8 +11,6 @@
 #include "secerr.h"
 #include "secport.h"
 
-typedef struct SECItemArrayStr SECItemArray;
-
 struct SECItemArrayStr {
     SECItem *items;
     unsigned int len;
