diff -up chromium-61.0.3163.79/third_party/WebKit/Source/platform/blob/BlobData.cpp.nonullptr chromium-61.0.3163.79/third_party/WebKit/Source/platform/blob/BlobData.cpp
--- chromium-61.0.3163.79/third_party/WebKit/Source/platform/blob/BlobData.cpp.nonullptr	2017-09-18 11:28:33.541339036 -0400
+++ chromium-61.0.3163.79/third_party/WebKit/Source/platform/blob/BlobData.cpp	2017-09-18 11:27:35.757799400 -0400
@@ -287,7 +287,7 @@ BlobDataHandle::BlobDataHandle(std::uniq
 
     size_t current_memory_population = 0;
     Vector<DataElementPtr> elements;
-    const DataElementPtr null_element = nullptr;
+    const DataElementPtr null_element;
     BlobBytesProvider* last_bytes_provider = nullptr;
 
     // TODO(mek): When the mojo code path is the default BlobData should
