diff -up chromium-5.0.358.0-20100318svn41971/src/chrome/browser/autofill/autofill_xml_parser.h.system-expat chromium-5.0.358.0-20100318svn41971/src/chrome/browser/autofill/autofill_xml_parser.h
--- chromium-5.0.358.0-20100318svn41971/src/chrome/browser/autofill/autofill_xml_parser.h.system-expat	2010-03-18 22:55:12.071283208 -0400
+++ chromium-5.0.358.0-20100318svn41971/src/chrome/browser/autofill/autofill_xml_parser.h	2010-03-18 22:55:27.671209061 -0400
@@ -10,7 +10,7 @@
 #include "base/basictypes.h"
 #include "chrome/browser/autofill/field_types.h"
 #include "chrome/browser/autofill/form_structure.h"
-#include "third_party/expat/files/lib/expat.h"
+#include <expat.h>
 #include "third_party/libjingle/source/talk/xmllite/xmlparser.h"
 
 // The base class that contains common functionality between
--- chromium-5.0.358.0-20100318svn41971/src/third_party/libjingle/source/talk/xmllite/xmlparser.h.system-expat	2010-03-18 22:55:12.071283208 -0400
+++ chromium-5.0.358.0-20100318svn41971/src/third_party/libjingle/source/talk/xmllite/xmlparser.h	2010-03-18 22:55:27.671209061 -0400
@@ -31,11 +31,7 @@
 #include <string>
 
 #include "talk/xmllite/xmlnsstack.h"
-#ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
-#else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
-#endif  // EXPAT_RELATIVE_PATH
+#include <expat.h>

 struct XML_ParserStruct;
 typedef struct XML_ParserStruct* XML_Parser;
--- chromium-5.0.358.0-20100318svn41971/src/third_party/libjingle/source/talk/xmllite/xmlbuilder.h.system-expat	2010-03-18 22:55:12.071283208 -0400
+++ chromium-5.0.358.0-20100318svn41971/src/third_party/libjingle/source/talk/xmllite/xmlbuilder.h	2010-03-18 22:55:27.671209061 -0400
@@ -33,11 +33,7 @@
 #include "talk/base/scoped_ptr.h"
 #include "talk/xmllite/xmlparser.h"
 
-#ifdef EXPAT_RELATIVE_PATH
-#include "expat.h"
-#else
-#include "third_party/expat/v2_0_1/Source/lib/expat.h"
-#endif  // EXPAT_RELATIVE_PATH
+#include <expat.h>
 
 namespace buzz {
 
