diff -up chromium-6.0.425.0-20100603svn48830/third_party/ffmpeg/ffmpeg.gyp.BAD chromium-6.0.425.0-20100603svn48830/third_party/ffmpeg/ffmpeg.gyp
--- chromium-6.0.425.0-20100603svn48830/src/third_party/ffmpeg/ffmpeg.gyp.BAD	2010-06-03 13:48:45.196313786 -0400
+++ chromium-6.0.425.0-20100603svn48830/src/third_party/ffmpeg/ffmpeg.gyp	2010-06-03 13:50:10.580325776 -0400
@@ -501,12 +501,14 @@
 
             # Add pkg-config result to include path when use_system_ffmpeg != 0
             ['use_system_ffmpeg != 0', {
-              'cflags': [
-                '<!@(pkg-config --cflags libavcodec libavformat libavutil)',
+              'include_dirs': [
+                'config',
+                'ffmpeg-0.6-headers',
               ],
               'direct_dependent_settings': {
-                'cflags': [
-                  '<!@(pkg-config --cflags libavcodec libavformat libavutil)',
+                'include_dirs': [
+                  'config',
+                  'ffmpeg-0.6-headers',
                 ],
               },
             }, {  # else use_system_ffmpeg == 0, add local copy to include path
