diff -urN chromium-56.0.2924.76.orig/remoting/host/security_key/fake_security_key_ipc_server.cc chromium-56.0.2924.76/remoting/host/security_key/fake_security_key_ipc_server.cc
--- chromium-56.0.2924.76.orig/remoting/host/security_key/fake_security_key_ipc_server.cc	2017-01-26 21:07:19.937348360 +0300
+++ chromium-56.0.2924.76/remoting/host/security_key/fake_security_key_ipc_server.cc	2017-01-26 21:10:34.481627009 +0300
@@ -130,7 +130,7 @@
 
   ipc_server_map_[connection_id] = fake_ipc_server->AsWeakPtr();
 
-  return fake_ipc_server;
+  return std::move(fake_ipc_server);
 }
 
 base::WeakPtr<FakeSecurityKeyIpcServer>
diff -urN chromium-56.0.2924.76.orig/third_party/webrtc/modules/desktop_capture/screen_capturer_x11.cc chromium-56.0.2924.76/third_party/webrtc/modules/desktop_capture/screen_capturer_x11.cc
--- chromium-56.0.2924.76.orig/third_party/webrtc/modules/desktop_capture/screen_capturer_x11.cc	2017-01-26 21:07:25.725445907 +0300
+++ chromium-56.0.2924.76/third_party/webrtc/modules/desktop_capture/screen_capturer_x11.cc	2017-01-26 21:11:22.865332765 +0300
@@ -412,7 +412,7 @@
     return nullptr;
   }
 
-  return capturer;
+  return std::move(capturer);
 }
 
 }  // namespace webrtc
