diff -up chromium-52.0.2743.82/third_party/WebKit/Source/wtf/text/TextCodecICU.cpp.morealiases chromium-52.0.2743.82/third_party/WebKit/Source/wtf/text/TextCodecICU.cpp
--- chromium-52.0.2743.82/third_party/WebKit/Source/wtf/text/TextCodecICU.cpp.morealiases	2016-08-03 08:55:48.293159940 +0200
+++ chromium-52.0.2743.82/third_party/WebKit/Source/wtf/text/TextCodecICU.cpp	2016-08-03 08:57:05.192505191 +0200
@@ -208,7 +208,11 @@ void TextCodecICU::registerEncodingNames
     registrar("iso_8859-6", "ISO-8859-6");
     registrar("iso_8859-7", "ISO-8859-7");
     registrar("iso_8859-8", "ISO-8859-8");
+    registrar("ISO_8859-8-I", "ISO-8859-8-I");
     registrar("iso_8859-9", "windows-1254");
+    registrar("ISO_8859-10", "ISO-8859-10");
+    registrar("ISO_8859-13", "ISO-8859-13");
+    registrar("ISO_8859-14", "ISO-8859-14");
     registrar("iso_8859-15", "ISO-8859-15");
     registrar("koi8_r", "KOI8-R");
     registrar("x-cp1253", "windows-1253");
