Index: build/gyp_chromium
diff --git a/build/gyp_chromium b/build/gyp_chromium
index e4cafc756134c35dc51c7ec3ea111bfef5157653..5c2644ecb374f280d0a775b319757352927b6ad7 100755
--- a/build/gyp_chromium
+++ b/build/gyp_chromium
@@ -39,7 +39,8 @@ sys.path.insert(1, os.path.join(chrome_src, 'third_party', 'liblouis'))
 sys.path.insert(1, os.path.join(chrome_src, 'third_party', 'WebKit',
     'Source', 'build', 'scripts'))
 
-import find_depot_tools
+if os.environ.get('GYP_GENERATORS') != 'make':
+  import find_depot_tools
 
 # On Windows, Psyco shortens warm runs of build/gyp_chromium by about
 # 20 seconds on a z600 machine with 12 GB of RAM, from 90 down to 70
@@ -474,8 +475,9 @@ if __name__ == '__main__':
     args.append('--check')
 
   supplemental_includes = GetSupplementalFiles()
-  if not RunGN(supplemental_includes):
-    sys.exit(1)
+  if os.environ.get('GYP_GENERATORS') != 'make':
+    if not RunGN(supplemental_includes):
+      sys.exit(1)
   args.extend(
       ['-I' + i for i in additional_include_files(supplemental_includes, args)])
 
