diff -urN chromium-54.0.2840.71.orig/ui/gfx/paint_vector_icon.cc chromium-54.0.2840.71/ui/gfx/paint_vector_icon.cc
--- chromium-54.0.2840.71.orig/ui/gfx/paint_vector_icon.cc	2016-10-31 02:03:25.121953757 +0300
+++ chromium-54.0.2840.71/ui/gfx/paint_vector_icon.cc	2016-10-31 02:08:53.022422171 +0300
@@ -317,6 +317,7 @@
                    VectorIconId badge_id)
       : CanvasImageSource(gfx::Size(dip_size, dip_size), false),
         id_(id),
+        path_(),
         color_(color),
         badge_id_(badge_id) {}
 
