--- native_client/build/download_toolchains.py	2013-09-20 17:28:08.000000000 +0200
+++ native_client/build/download_toolchains.py	2013-09-23 18:17:31.760826152 +0200
@@ -235,11 +235,11 @@
     filepath = os.path.join(download_dir, '.'.join([filepath, ext]))
     filepaths.append(filepath)
     # If we did not need to synchronize, then we are done
-    if download_utils.SyncURL(url, filepath, stamp_dir=stamp_dir,
-                              min_time=min_time, hash_val=hash_val,
-                              stamp_index=index,
-                              keep=keep, verbose=verbose):
-      need_sync = True
+    # if download_utils.SyncURL(url, filepath, stamp_dir=stamp_dir,
+    #                         min_time=min_time, hash_val=hash_val,
+    #                         stamp_index=index,
+    #                         keep=keep, verbose=verbose):
+    need_sync = True
     index += 1
 
   if not need_sync:
